lib/rubocop/cop/rails/reflection_class_name_string.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks if the value of the option `class_name`, in
+      # the definition of a reflection is a string.
+      #
+      # @example
+      #   # bad
+      #   has_many :accounts, class_name: Account
+      #   has_many :accounts, class_name: Account.name
+      #
+      #   # good
+      #   has_many :accounts, class_name: 'Account'
+      class ReflectionClassNameString < Cop
+        MSG = 'Use a string has value for a class_name.'.freeze
+
+        def_node_matcher :association_with_options?, <<-PATTERN
+          (send nil? {:has_many :has_one :belongs_to} _ (hash $...))
+        PATTERN
+
+        def_node_search :class_name_string, <<-PATTERN
```

Comment:
Does this mean `:reflection_class_name` node pattern?

Reply:
This pattern match for the key (`class_name`) and his value of a hash. The reflection matcher itself is defined [here](https://github.com/rubocop-hq/rubocop/pull/6704/files/7693b518d72dc4b8f7e6584f0b119b0db3d203e1#diff-21df1c369efb890c11c7d6ecba1911f0R19).

Reply:
This s-expression pattern seems to match a pattern using constants like `class_name: Account`, for example. This represents a case where the value is not a string (`class_name: 'Account'`).
In other words it seems to match a reflection class name other than a string class name.

Reply:
I see what you mean. It's true that the name of the pattern is confusing. I can change it for something like `class_name_not_string`. 

Suggestions are welcome.

Reply:
I like `reflaction_class_name` as I think of it.

The current code base change when using this is as follows.

```diff
% g diff
diff --git a/lib/rubocop/cop/rails/reflection_class_name_string.rb b/lib/rubocop/cop/rails/reflection_class_name_string.rb
index c3dfa96c5..542a89f7e 100644
--- a/lib/rubocop/cop/rails/reflection_class_name_string.rb
+++ b/lib/rubocop/cop/rails/reflection_class_name_string.rb
@@ -20,17 +20,17 @@ module RuboCop
           (send nil? {:has_many :has_one :belongs_to} _ (hash $...))
         PATTERN

-        def_node_search :class_name_string, <<-PATTERN
+        def_node_search :reflection_class_name, <<-PATTERN
           (pair (sym :class_name) !str)
         PATTERN

         def on_send(node)
-          return unless association_with_options? node
+          return unless association_with_options?(node)

-          class_name_node = class_name_string(node).first
-          return unless class_name_node
+          reflection_class_name = reflection_class_name(node).first
+          return unless reflection_class_name

-          add_offense(node, location: class_name_node.loc.expression)
+          add_offense(node, location: reflection_class_name.loc.expression)
         end
       end
     end
```

"`return unless reflaction_class_name`" can represent the intent to reach `add_offense` when a node matches the reflaction class name.